src/argilla/client/sdk/datasets/api.py

```diff
@@ -38,6 +38,28 @@ def get_dataset(client: AuthenticatedClient, name: str) -> Response[Dataset]:
     return _build_response(response=response, name=name)
 
 
+def list_datasets(client: AuthenticatedClient, workspace: Optional[str] = None) -> Response[List[Dataset]]:
+    url = "{}/api/datasets".format(client.base_url)
+
+    response = httpx.get(
+        url=url,
+        params={"workspace": workspace} if workspace else None,
+        headers=client.get_headers(),
+        cookies=client.get_cookies(),
+        timeout=client.get_timeout(),
+    )
+
+    if response.status_code == 200:
+        parsed_response = [Dataset(**dataset) for dataset in response.json()]
+        return Response(
+            status_code=response.status_code,
+            content=response.content,
+            headers=response.headers,
+            parsed=parsed_response,
+        )
```

Comment:
```suggestion
        response_obj = Response.from_httpx_response(response)
        response_obj.parsed = [Dataset(**dataset) for dataset in response.json()]
        return response_obj
```

Reply:
Makes sense, didn't do it because this is low priority and changing something would imply changing the whole file (if not every single file that it's outdated)